File: ConcreteNg/ConcreteNg.Shared/Models/TableRequest.cs
using ConcreteNg.Shared.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConcreteNg.Shared.Models
{
    public class TableRequest
    {
        public int CurrentPage { get; set; }
        public int PageSize { get; set; }
        public string OrderBy { get; set; }
        public bool IsAscending { get; set; }
        public IEnumerable<BaseFilter> Filters { get; set; }

    }

    public class BaseFilter
    {
        public int ColumnName { get; set; }
        public string FilterQuery { get; set; }
    }
}

File: ConcreteNg.WebApp/src/app/models/table-request.ts
import { DiaryFilterColumnsEnum } from "../enums/diary-filter-columns-enum"
import { PartnerListFilterEnum } from "../enums/partner-list-filter-enum"
import { PricingListFilterColumnsEnum } from "../enums/pricing-list-filter-columns-enum"
import { ProjectFilterColumnsEnum } from "../enums/project-filter-columns-enum"
import { UserListFilterEnum } from "../enums/user-list-filter-enum"

export interface TableRequest {
    currentPage: number,
    pageSize: number,
    orderBy: string,
    isAscending: boolean,
    filters: BaseFilter[]
}

export interface BaseFilter{
    columnName: ProjectFilterColumnsEnum | PricingListFilterColumnsEnum | DiaryFilterColumnsEnum | PartnerListFilterEnum | UserListFilterEnum,
    filterQuery: string
}
